### PR TITLE
Please review my PR for fyle django oauth template

### PR DESCRIPTION
**Pull request by subhajit21**
Client Id ,Client Secret and base_url for FyleSDK connection are accepting value by Environment variable .Others necessary URLs are hard coded . Hardcoded URLs can be removed and can accept value by Environment Variable . Displaying Employee data in basic HTML page .
Please review my PR.

### CODE REVIEW

File: Djangotempalate/settings.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+# Build paths inside the project like this: os.path.join(BASE_DIR, ...)
+BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+
+
+# Quick-start development settings - unsuitable for production
+# See https://docs.djangoproject.com/en/2.2/howto/deployment/checklist/
+
+# SECURITY WARNING: keep the secret key used in production secret!
+SECRET_KEY = '#u_1%o*b1b$51m9$@%v_p*@+u@pfwx0irk=2pcproa*2$xus()'
```
Comment (Shwetabhk) [inline Djangotempalate/settings.py]: Take secret key from the env

File: fyleOauthDjango/views.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        elif error == "access_denied":
+            return redirect('/login')
+        return self.authorize(self.request)
+
+    def authorize(self,request):
+        global BASE_URL
+        global CLIENT_ID
+        protocol = "http://"
+        domain = request.get_host()
+        redirect_uri = "{0}{1}/profile".format(protocol, domain)
```
Comment (Shwetabhk) [inline fyleOauthDjango/views.py]: dont use variables for protocol and domain

File: fyleOauthDjango/connectors/fyle.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        self.__refresh_token = refresh_token
+
+        self.__connection = FyleSDK(
+            base_url=self.__base_url,
+            client_id=self.__client_id,
+            client_secret=self.__client_secret,
+            refresh_token=self.__refresh_token
+        )
+
+    def get_employee_details(self):
```
Comment (Shwetabhk) [inline fyleOauthDjango/connectors/fyle.py]: call this `extract_employee_details`:

File: Djangotempalate/wsgi.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+For more information on this file, see
+https://docs.djangoproject.com/en/2.2/howto/deployment/wsgi/
+"""
+
+import os
+
+from django.core.wsgi import get_wsgi_application
+
+os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'Djangotempalate.settings')
```
Comment (Shwetabhk) [inline Djangotempalate/wsgi.py]: template spelling is wrong

File: fyleOauthDjango/apps.py [hunk 5]
```diff
@@ -0,0 +1,5 @@
+from django.apps import AppConfig
+
+
+class FyleoauthdjangoConfig(AppConfig):
```
Comment (Shwetabhk) [inline fyleOauthDjango/apps.py]: Use camel case `FyleOauthDjangoConfig`

File: fyleOauthDjango/views.py [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+from django.http import HttpResponseRedirect
+from django.shortcuts import render, redirect
+from django.views.generic import TemplateView
+
+from .connectors import fyle
+
+CLIENT_ID = os.environ.get("CLIENT_ID")
+CLIENT_SECRET = os.environ.get("CLIENT_SECRET")
+BASE_URL = os.environ.get("BASE_URL")
+TOKEN_URL = '{0}/api/oauth/token'.format(BASE_URL)
```
Comment (Shwetabhk) [inline fyleOauthDjango/views.py]: If it is used only once, no need to make it a global variable

File: Djangotempalate/settings.py [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+    'django.middleware.security.SecurityMiddleware',
+    'django.contrib.sessions.middleware.SessionMiddleware',
+    'django.middleware.common.CommonMiddleware',
+    'django.middleware.csrf.CsrfViewMiddleware',
+    'django.contrib.auth.middleware.AuthenticationMiddleware',
+    'django.contrib.messages.middleware.MessageMiddleware',
+    'django.middleware.clickjacking.XFrameOptionsMiddleware',
+]
+
+ROOT_URLCONF = 'Djangotempalate.urls'
```
Comment (Shwetabhk) [inline Djangotempalate/settings.py]: template spelling is wrong

File: fyleOauthDjango/templates/welcome.html [hunk 8]
```diff
@@ -0,0 +1,19 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <title>Welcome Django template</title>
```
Comment (Shwetabhk) [inline fyleOauthDjango/templates/welcome.html]: just make it welcome

File: fyleOauthDjango/views.py [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+        elif error == "access_denied":
+            return redirect('/login')
+        return self.authorize(self.request)
+
+    def authorize(self, request):
+        global BASE_URL
+        global CLIENT_ID
+        redirect_uri = "{0}{1}/profile".format("http://", request.get_host())
+        authorize_url = "{0}/app/main/#/oauth/authorize?client_id={1}&redirect_uri={" \
+                        "2}&response_type=code&state=ajsfbjak".format(
```
Comment (Shwetabhk) [inline fyleOauthDjango/views.py]: move `2}` in the above line

File: fyleOauthDjango/apps.py (hunk 5 above)
Comment (subhajit21) [reply fyleOauthDjango/apps.py]: `FyleoauthdjangConfig` is autogenerated name. you can see in my application name , I used camel case. I did not write this class name , It is autogenerated.

**Pull request merged**